lib/kube/proxy/server.go

```diff
@@ -316,6 +322,25 @@ func (t *TLSServer) getServerInfo(name string) (types.Resource, error) {
 	return srv, nil
 }
 
+// getSafeKubeClusterWithServiceLabels finds the kube cluster by name, strips the credentials
+// for Azure, AWS and Kubeconfig in order to avoid sharing them via heartbeat and
+// adds the service static and dynamic labels.
+func (t *TLSServer) getSafeKubeClusterWithServiceLabels(name string) (*types.KubernetesClusterV3, error) {
```

Comment:
Tbh I find the way the labels are currently managed quite confusing as part of it is here in the server but it actually calls out to the forwarder so it's a little hard to understand how labels are actually set.

Can it be cleaned up a bit? For example, why does the forwarder have to deal with things like labels, it only needs to forward requests and not handle labels that get reported as a part of the heartbeat imo.

Can this kube/proxy/server.go:TLSServer be initialized with the service's static/dynamic labels and update them on the clusters that are being heartbeat, without splitting this logic between server/forwarder? That would be closer to how it's done in db/app services too.

Reply:
I can refactor it. Previously service static and dynamic labels were part of the Forwarder config